app/src/main/java/com/codesoom/assignment/domain/CatToy.java

```diff
@@ -0,0 +1,146 @@
+package com.codesoom.assignment.domain;
+
+import com.codesoom.assignment.exception.CatToyInvalidPriceException;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.util.Objects;
+
+/**
+ * 고양이 장난감 객체로 장난감에 대한 정보를 담는다.
+ */
+
+@Entity
+public class CatToy {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+    private String maker;
+    private Long price;
+    private String imageUrl;
+
+    public CatToy() {}
+
+    public CatToy(String name, String maker, Long price, String imageUrl) {
+       this(null, name, maker, price, imageUrl);
+    }
+
+    public CatToy(Long id, String name, String maker, Long price, String imageUrl) {
+        isValidPrice(price);
+
+        this.id = id;
+        this.name = name;
+        this.maker = maker;
+        this.price = price;
+        this.imageUrl = imageUrl;
+    }
+
+    private void isValidPrice(Long price) {
+        if (price < 0) {
+            throw new CatToyInvalidPriceException(price);
+        }
+    }
```

Comment:
상식에 어긋납니다. 이 메소드의 시그니처를 보면 올바른 가격인지 아닌지를 물어보는 관례적인 이름을 갖고 있어요. 그렇다면 올바른 가격이면 true, 그렇지 않다면 false를 리턴하면 됩니다. 올바른지 물어봤는데 다짜고짜 예외를 던지면 이건 goto 문이나 다름 없습니다. 어딘가에 있을 어드바이스로 goto 하는 거죠. 다음 두 방법 중 하나를 고려해 보세요.

- boolean을 리턴하게 한다.
- 메소드 이름을 변경한다.

```suggestion
    private void isValidPrice(Long price) {
        return price >= 0;
    }
```

Reply:
객체지향 생활 체조의 들여쓰기 레벨 2를 넘기지 말라는 제약 때문에 indent를 위해 호출하는 메서드에서 바로 예외를 호출하게 할 것인가. 아니면 예외처리를 스스로 처리하게 할 것인가에 대한 트레이드 오프는 그 때 그 때 고려해서 결정해야 할 문제라고 생각하겠습니다! 

마냥 들여쓰기만 고민하다보니 한 쪽에 치우친 코드가 된 것 같네요. 그리고 메서드명도 `is`라는 접두사를 보면 진리값을 반환할 것 같이 해놓고 void 타입인것도 다시 보니 문제가 있네요. 수정했습니다 👍